core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -732,6 +732,11 @@ default boolean isMaterializedView(Session session, QualifiedObjectName viewName
      */
     boolean supportsReportingWrittenBytes(Session session, QualifiedObjectName tableName, Map<String, Object> tableProperties);
 
+    /**
+     * Returns true if the connector reports number of written bytes for a new table. Otherwise, it returns false.
+     */
+    boolean supportsMultipleWriterPerPartition(Session session, QualifiedObjectName schemaTableName);
```

Comment:
What does “partition” mean in this context? The engine has no such concept, at least in the sense of hive/iceberg/delta lake partitions. We should not be leaking that through the abstraction.